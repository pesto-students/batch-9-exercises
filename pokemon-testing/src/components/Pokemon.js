import React from 'react';
import PropTypes from 'prop-types';

const Pokemon = ({ pokemon }) => (
  <li className="pokemons__item">
    <button
      type="button"
      className="pokemons__sprite"
      style={{
        backgroundImage: `url(${`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemon.id}.png`})`,
      }}
    />
    <p className="pokemons__name">{pokemon.name}</p>
  </li>
);

Pokemon.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default Pokemon;
