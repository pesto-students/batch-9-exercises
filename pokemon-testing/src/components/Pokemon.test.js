import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pokemon from './Pokemon';

describe('<Pokemon/>', () => {
  test('', () => {
    expect(true).toBe(true);
  });

  test('matches the snapshot', () => {
    const tree = shallow(
      <Pokemon
        pokemon={{
          name: 'bulbasaur',
          url: 'https://pokeapi.co/api/v2/pokemon/1/'
        }}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
