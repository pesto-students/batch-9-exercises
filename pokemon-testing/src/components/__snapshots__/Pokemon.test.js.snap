// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon/> matches the snapshot 1`] = `
<li
  className="pokemons__item"
>
  <button
    className="pokemons__sprite"
    style={
      Object {
        "backgroundImage": "url(https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png)",
      }
    }
    type="button"
  />
  <p
    className="pokemons__name"
  >
    bulbasaur
  </p>
</li>
`;
